### PR TITLE
Update Jackson, Netty, OkHttp, and Reactor Dependencies

### PR DESCRIPTION
**Pull request by alzimmermsft**

### CODE REVIEW

File: eng/versioning/external_dependencies.txt [hunk 1]
```diff
@@ -7,14 +7,14 @@
...
-com.fasterxml.jackson.datatype:jackson-datatype-jsr310;2.12.2
-com.fasterxml.jackson.module:jackson-module-parameter-names;2.12.2
+com.fasterxml.jackson.core:jackson-annotations;2.12.3
+com.fasterxml.jackson.core:jackson-core;2.12.3
+com.fasterxml.jackson.core:jackson-databind;2.12.3
+com.fasterxml.jackson.dataformat:jackson-dataformat-avro;2.12.3
+com.fasterxml.jackson.dataformat:jackson-dataformat-xml;2.12.3
+com.fasterxml.jackson.datatype:jackson-datatype-jdk8;2.12.3
+com.fasterxml.jackson.datatype:jackson-datatype-jsr310;2.12.3
+com.fasterxml.jackson.module:jackson-module-parameter-names;2.12.3
```
Comment (chenrujun) [inline eng/versioning/external_dependencies.txt:17]: Hi, @alzimmermsft 
I think we should keep these versions same to spring-boot.
Refs:
1. https://github.com/Azure/azure-sdk-for-java/pull/19968/files
2. https://github.com/Azure/azure-sdk-for-java/pull/20010/files

Comment (alzimmermsft) [reply eng/versioning/external_dependencies.txt:17]: Keep this to `2.12.2`, correct? I'll change this next commit.

Comment (chenrujun) [reply eng/versioning/external_dependencies.txt]: I'll create a PR to upgrade spring-boot to 2.4.5 today. And update external dependencies' version.

Comment (chenrujun) [reply eng/versioning/external_dependencies.txt]: PR created: https://github.com/Azure/azure-sdk-for-java/pull/20721/files

The version is still `2.12.3`. 

>Keep this to 2.12.2, correct? I'll change this next commit.

Yes, please keep to `2.12.2`.

Let's upgrade to `2.12.3` after` spring-boot-2.5.0` released.

**Comment by alzimmermsft** [comment]
Closing in favor of #20721

File: eng/versioning/external_dependencies.txt (hunk 1 above)
Comment (chenrujun) [reply eng/versioning/external_dependencies.txt]: And here is the PR to upgrade to spring-boot-2.5.0-RC1: https://github.com/Azure/azure-sdk-for-java/pull/20010/files#diff-3813fa7f30abaf4f8f3f06486d039e8beee30250fef5c7ddb8de42eda07fd300L10

**Pull request closed without merging**